inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -68,90 +66,74 @@ def arxiv_fulltext_download(obj, eng):
+        url=current_app.config['ARXIV_TARBALL_URL'].format(
+            arxiv_id=arxiv_id
         )
+    )
+    if tarball:
+        tarball['doctype'] = "package"
+        obj.log.info("Tarball retrieved from arXiv for {0}".format(arxiv_id))
     else:
-        tarball = obj.files[filename]
-
-    try:
-        plots = process_tarball(tarball.file.uri)
-    except (InvalidTarball, NoTexFilesFound):
-        obj.log.error(
-            'Invalid tarball {0}'.format(tarball.file.uri)
-        )
-        return
-    except DelegateError as err:
-        obj.log.error("Error extracting plots. Report and skip.")
-        current_app.logger.exception(err)
-        return
-
-    for idx, plot in enumerate(plots):
-        obj.files[plot.get('name')] = BytesIO(open(plot.get('url')))
-        obj.files[plot.get('name')]["doctype"] = "Plot"
-        obj.files[plot.get('name')]["description"] = u"{0:05d} {1}".format(
-            idx, "".join(plot.get('captions', []))
-        )
-    obj.log.info("Added {0} plots.".format(len(plots)))
+        obj.log.error("Cannot retrieve tarball from arXiv for {0}".format(arxiv_id))
 
 
 @with_debug_logging
-def arxiv_refextract(obj, eng):
-    """Extract references from arXiv PDF.
+def arxiv_plot_extract(obj, eng):
+    """Extract plots from an arXiv archive.
 
     :param obj: Workflow Object to process
     :param eng: Workflow Engine processing the object
     """
+    from wand.exceptions import DelegateError
+
     arxiv_id = get_arxiv_id(obj.data)
-    filename = secure_filename("{0}.pdf".format(arxiv_id))
-    if filename not in obj.files:
-        pdf = download_file_to_workflow(
-            workflow=obj,
-            name=filename,
-            url=current_app.config['ARXIV_PDF_URL'].format(
-                arxiv_id=arxiv_id
+    filename = secure_filename("{0}.tar.gz".format(arxiv_id))
+    tarball = obj.files[filename]
+
+    if tarball:
+        try:
+            plots = process_tarball(tarball.file.uri)
+        except (InvalidTarball, NoTexFilesFound):
+            obj.log.error(
+                'Invalid tarball {0}'.format(tarball.file.uri)
             )
-        )
-    else:
-        pdf = obj.files[filename]
-    if pdf:
-        mapped_references = extract_references(pdf.file.uri)
-        if mapped_references:
-            obj.data["references"] = mapped_references
-            obj.log.info("Extracted {0} references".format(
-                len(mapped_references)
-            ))
-        else:
-            obj.log.info("No references extracted")
-    else:
-        obj.log.error("Not able to download and process the PDF")
+            return
+        except DelegateError as err:
+            obj.log.error("Error extracting plots. Report and skip.")
+            current_app.logger.exception(err)
+            return
+
+        for idx, plot in enumerate(plots):
+            obj.files[plot.get('name')] = BytesIO(open(plot.get('url')))
+            obj.files[plot.get('name')]["doctype"] = "Plot"
```

Comment:
Yeah but I can't correct everything in this PR. We can add it as an issue for later...